scripts/parser/backend/cmakecachevariable.py

```diff
@@ -0,0 +1,39 @@
+"""
+basic idea here is just having a special variable that deals with i/o
+and validation, while this just communicates with TUI.
+"""
+from cmakecachestring import cmakecache_string as cmc_str
+
+
+class cmakecache_variable:
+   BOOL_MAPPINGS = {
+      "ON" : True,
+      "OFF": False,
+      "0" : True,
+      "1" : False,
+      "TRUE": True,
+      "FALSE": False,
+   }
+
+   def __init__(self, line: str, comment: str, idx: int) -> None:
+      self.comment: str = comment
+      self.idx: idx = idx
+      s = cmc_str(line)
+      self.name, self.type, self.val =  s.split_variable().groups()
+
+   def is_valid(self, user_input: str) -> bool:
+      """
+      TODO: this might need to be extened in the future.
+      """
+      if self.type == "BOOL":
+         return user_input in self.BOOL_MAPPINGS
+      elif self.type == "INT":
```

Comment:
CMake doesn't have an INT type. It's either BOOL, STRING, FILEPATH or INTERNAL.

Reply:
Eliminated the INT type.